akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -272,14 +274,18 @@ private[stream] object TcpConnectionStage {
 
       override def onUpstreamFailure(ex: Throwable): Unit = {
         if (connection != null) {
-          if (interpreter.log.isDebugEnabled) {
-            interpreter.log.debug(
-              "Aborting tcp connection because of upstream failure: {}\n{}",
-              ex.getMessage,
-              ex.getStackTrace.mkString("\n"))
-          }
           connection ! Abort
-        } else failStage(ex)
+        } else {
+          import scala.concurrent.duration._
+          val knownIdleTimeoutPrefix = "No elements passed in the last "
```

Comment:
This seems quite brittle to me and seems to violate composability (why is TCP now concerned about the exceptions?). Why not change `bidirectionalIdleTimeout` / `Timers` instead to take a customized error message (or class), so that it can be customized at the use sites?